arraycontext/impl/pytato/fake_numpy.py

```diff
@@ -91,22 +95,50 @@ def minimum(self, x, y):
     def where(self, criterion, then, else_):
         return rec_multimap_array_container(pt.where, criterion, then, else_)
 
-    def sum(self, a, axis=None, dtype=None):
-        def _pt_sum(ary):
+    @staticmethod
+    def _reduce(container_binop, array_reduce,
+            ary, *,
+            axis, dtype, initial):
+        def container_reduce(ctr):
+            if initial is _NoValue:
+                try:
+                    return reduce(container_binop, ctr)
+                except TypeError as exc:
+                    assert "empty sequence" in str(exc)
+                    raise ValueError("zero-size reduction operation "
+                            "without supplied 'initial' value")
```

Comment:
Should this maybe be
```suggestion
                    if "empty sequence" in str(exc):
                        raise ValueError("zero-size reduction operation "
                                "without supplied 'initial' value")
                    else:
                        raise        
```
? (In case an unrelated `TypeError` is raised.)